src/lib.rs

```diff
@@ -304,13 +304,10 @@ impl WalkDir {
     /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
     ///     a.cmp(b)
     /// }
-    /// WalkDir::new("foo").sort_by(Some(string_compare));
+    /// WalkDir::new("foo").sort_by(string_compare);
```

Comment:
Could you use a closure here? Before, you needed the `fn`, but now it'd be much nicer to write `sort_by(|a, b| a.cmp(b))`.